lib/new_relic/agent/distributed_tracing/cross_app_tracing.rb

```diff
@@ -53,6 +52,9 @@ def cat_path_hash
 
       def insert_cross_app_header(headers)
         return unless CrossAppTracing.cross_app_enabled?
+        # do not insert CAT headers for gRPC requests
+        # https://github.com/newrelic/newrelic-ruby-agent/issues/1730
+        return if grpc_headers?(headers)
```

Comment:
It looks like "then" case coverage is missing for this line.

Reply:
Odd. There was an existing test for the presence of the headers that should continue to test the `return` not being reached, and there has been a new test added to test the `return` being reached. Both sides of the `if` conditional should be getting tested. I'll revisit after I move the check as Tanna is suggesting.

Reply:
Moot with the [26e4415](https://github.com/newrelic/newrelic-ruby-agent/pull/1837/commits/26e4415437bb1e86f455b45bedc947507a7b0b93) rework.